walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,48 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        Collection<TimeRange> availableTimeRange = new ArrayList<TimeRange>();
+        
+        if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
```

Comment:
One way to improve if condition is to use a method in order to easily explain and encapsulate complexity, e.g. `if (isMeetingRequestTooLong(...))`